includes/admin/load.php

```diff
@@ -204,137 +207,142 @@ function perflab_dismiss_wp_pointer_wrapper() {
  * Callback function to handle admin scripts.
  *
  * @since 2.8.0
+ * @since n.e.x.t Renamed to perflab_enqueue_features_page_scripts().
  */
-function perflab_enqueue_modules_page_scripts() {
-	wp_enqueue_script( 'updates' );
-
-	wp_localize_script(
-		'updates',
-		'_wpUpdatesItemCounts',
-		array(
-			'settings' => array(
-				'totals' => wp_get_update_data(),
-			),
-		)
-	);
-
+function perflab_enqueue_features_page_scripts() {
 	wp_enqueue_script( 'thickbox' );
 	wp_enqueue_style( 'thickbox' );
 
 	wp_enqueue_script( 'plugin-install' );
-
-	wp_enqueue_script(
-		'perflab-plugin-management',
-		plugin_dir_url( __FILE__ ) . 'js/perflab-plugin-management.js',
-		array( 'jquery' ),
-		'1.0.0',
-		array(
-			'in_footer' => true,
-			'strategy'  => 'defer',
-		)
-	);
 }
 
 /**
- * Callback function hooked to admin_action_perflab_activate_plugin to handle plugin activation.
+ * Callback for handling installation/activation of plugin.
  *
- * @since 2.8.0
+ * @since n.e.x.t
  */
-function perflab_activate_plugin() {
-	// Do not proceed if plugin query arg is not present.
-	if ( empty( $_GET['plugin'] ) ) {
-		return;
+function perflab_install_activate_plugin_callback() {
+	check_admin_referer( 'perflab_install_activate_plugin' );
+
+	require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-ajax-upgrader-skin.php';
+
+	if ( ! isset( $_GET['slug'] ) ) {
+		wp_die( esc_html__( 'Missing required parameter.', 'performance-lab' ) );
 	}
 
-	$plugin = sanitize_text_field( wp_unslash( $_GET['plugin'] ) );
+	$plugin_slug = sanitize_text_field( wp_unslash( $_GET['slug'] ) );
 
-	check_admin_referer( "perflab_activate_plugin_{$plugin}" );
+	if ( ! $plugin_slug ) {
+		wp_die( esc_html__( 'Invalid plugin.', 'performance-lab' ) );
+	}
+
+	$is_plugin_installed = ! isset( $_GET['file'] ) || ! $_GET['file'];
+
+	// Install the plugin if it is not installed yet.
+	if ( $is_plugin_installed ) {
+		// Check if the user have plugin installation capability.
+		if ( ! current_user_can( 'install_plugins' ) ) {
+			wp_die( esc_html__( 'Sorry, you are not allowed to manage plugins for this site. Please contact the administrator.', 'performance-lab' ) );
+		}
+
+		$api = perflab_query_plugin_info( $plugin_slug );
+
+		// Return early if plugin API returns an error.
+		if ( ! $api ) {
+			wp_die( esc_html__( 'An unexpected error occurred. Something may be wrong with WordPress.org or this server&#8217;s configuration.', 'performance-lab' ) );
```

Comment:
Same here, we can make use of a core string instead.
```suggestion
			wp_die(
				wp_kses(
					sprintf(
						/* translators: %s: Support forums URL. */
						__( 'An unexpected error occurred. Something may be wrong with WordPress.org or this server&#8217;s configuration. If you continue to have problems, please try the <a href="%s">support forums</a>.', 'default' ),
						__( 'https://wordpress.org/support/forums/', 'default' )
					),
					array( 'a' => array( 'href' => true ) )
				)
			);
```